code-studio/src/js/pairing.js

```diff
@@ -0,0 +1,32 @@
+/* global dashboard, $ */
+
+import ReactDOM from 'react-dom';
+import PairingDialog from './components/pairing/pairing_dialog.jsx';
+
+export default {
+  init(pairingUrl, hideUserOptions, showPairingDialog) {
+    let pairingDialog;
+
+    function lazyInitDialog() {
+      if (!pairingDialog) {
+        $('body').append('<div id="pairing"></div>');
+
+        pairingDialog = ReactDOM.render(
+          <PairingDialog source={pairingUrl} />,
+          document.getElementById('pairing')
```

Comment:
nit: in the absurd case that there's another element with the id 'pairing', it would be marginally better to do something like
```javascript
let container = document.createElement('div');
container.id = pairing
$('body').append(container)
pairingDialog = ReactDOM.render(<PairingDialog source={pairingUrl} />, container);
```